pkg/gateway/model_build_servicenetwork.go

```diff
@@ -63,25 +70,33 @@ func (t *serviceNetworkModelBuildTask) buildModel(ctx context.Context) error {
 
 func (t *serviceNetworkModelBuildTask) buildServiceNetwork(ctx context.Context) error {
 	spec := latticemodel.ServiceNetworkSpec{
-		Name:           t.gateway.Name,
-		Namespace:      t.gateway.Namespace,
-		Account:        config.AccountID,
-		AssociateToVPC: false,
+		Name:      t.gateway.Name,
+		Namespace: t.gateway.Namespace,
+		Account:   config.AccountID,
 	}
 
-	// by default it is true
-	spec.AssociateToVPC = true
+	// default associateToVPC is true
+	associateToVPC := true
 	if len(t.gateway.ObjectMeta.Annotations) > 0 {
 		if value, exist := t.gateway.Annotations[LatticeVPCAssociationAnnotation]; exist {
 			if value == "true" {
-				spec.AssociateToVPC = true
+				associateToVPC = true
 			} else {
-				spec.AssociateToVPC = false
+				associateToVPC = false
 			}
+		}
+	}
 
+	if t.vpcAssociationPolicy != nil {
+		if t.vpcAssociationPolicy.Spec.AssociateWithVpc != nil {
+			associateToVPC = *t.vpcAssociationPolicy.Spec.AssociateWithVpc
 		}
 
+		if t.vpcAssociationPolicy.Spec.SecurityGroupIds != nil {
+			spec.SecurityGroupIds = securityGroupIdsToStringPointersSlice(t.vpcAssociationPolicy.Spec.SecurityGroupIds)
```

Comment:
you can use `aws.StringSlice` or `aws.StringSliceValue` here to convert from `[]string` to `[]*string` and vice versa

Reply:
I found the ` aws.StringSlice(src []string)` can only accept `[]string` input param,  but the  t.vpcAssociationPolicy.Spec.SecurityGroupIds is actually `[]SecurityGroupId ` cannot use that method